chainerx_cc/chainerx/routines/indexing.cc

```diff
@@ -166,17 +169,15 @@ Array TakeOp::Call(const Array& a, const Array& indices, int8_t axis) {
 
     {
         NoBackpropModeScope scope{};
-        Impl(a, indices, axis_norm, out);
+        a.device().backend().CallOp<TakeOp>(a, indices, axis_norm, out);
     }
 
     BackwardBuilder bb{"take", a, out};
     if (BackwardBuilder::Target bt = bb.CreateTarget(0)) {
         CHAINERX_ASSERT(internal::GetArrayBody(indices)->nodes().empty());
         bt.Define([indices, axis_norm, a_shape = a.shape()](BackwardContext& bctx) {
             const Array& gout = *bctx.output_grad();
-            Array gx = Zeros(a_shape, gout.dtype(), gout.device());
-            AddAt(gx, indices, axis_norm, gout, gx);
-            bctx.input_grad() = std::move(gx);
+            bctx.input_grad() = AddAt(Zeros(a_shape, gout.dtype(), gout.device()), indices, axis_norm, gout);
```

Comment:
Isn't this suboptimal because now we allocate the buffer for output twice?

Reply:
https://github.com/chainer/chainer/pull/6789